drivers/mspi/mspi_stm32_xspi.c

```diff
@@ -0,0 +1,1879 @@
+		data->dev_cfg.time_to_break = dev_cfg->time_to_break;
+	}
+	if ((param_mask & MSPI_DEVICE_CONFIG_CE_POL) != 0) {
+		ret = mspi_stm32_xspi_validate_ce_polarity(dev_cfg->ce_polarity);
+		if (ret == 0) {
+			goto end;
+		}
+		data->dev_cfg.ce_polarity = dev_cfg->ce_polarity;
+	}
+
+	if ((param_mask & MSPI_DEVICE_CONFIG_DQS) != 0) {
+		ret = mspi_stm32_xspi_validate_dqs(dev_cfg->dqs_enable, cfg->mspicfg.dqs_support);
+		if (ret != 0) {
+			goto end;
+		}
+		data->dev_cfg.dqs_enable = dev_cfg->dqs_enable;
+	}
+
+	if ((param_mask & MSPI_DEVICE_CONFIG_ENDIAN) != 0) {
+		ret = mspi_stm32_xspi_validate_endian(dev_cfg->endian);
+		if (ret != 0) {
+			goto end;
+		}
+		data->dev_cfg.endian = dev_cfg->endian;
+	}
+end:
+	return ret;
+}
+
+/**
+ * Check and save dev_cfg to controller data->dev_cfg.
+ *
+ * @param controller Pointer to the device structure for the driver instance.
+ * @param param_mask Macro definition of what to be configured in cfg.
+ * @param dev_cfg The device runtime configuration for the MSPI controller.
+ * @return 0 MSPI device configuration successful.
+ * @return -Error MSPI device configuration fail.
+ */
+static int mspi_stm32_xspi_dev_cfg_save(const struct device *controller,
+					const enum mspi_dev_cfg_mask param_mask,
+					const struct mspi_dev_cfg *dev_cfg)
+{
+	const struct mspi_stm32_conf *cfg = controller->config;
+	struct mspi_stm32_data *data = controller->data;
+	int ret = 0;
+
+	if ((param_mask & MSPI_DEVICE_CONFIG_CE_NUM) != 0) {
+		data->dev_cfg.ce_num = dev_cfg->ce_num;
+	}
+
+	if ((param_mask & MSPI_DEVICE_CONFIG_FREQUENCY) != 0) {
+		ret = mspi_stm32_xspi_validate_freq(dev_cfg->freq);
+		if (ret != 0) {
+			goto end;
+		}
+		data->dev_cfg.freq = dev_cfg->freq;
+	}
+
+	if ((param_mask & MSPI_DEVICE_CONFIG_IO_MODE) != 0) {
+		ret = mspi_stm32_xspi_validate_io_mode(dev_cfg->io_mode);
+		if (ret != 0) {
+			goto end;
+		}
+		data->dev_cfg.io_mode = dev_cfg->io_mode;
+	}
+
+	if ((param_mask & MSPI_DEVICE_CONFIG_DATA_RATE) != 0) {
+		ret = mspi_stm32_xspi_validate_data_rate(dev_cfg->data_rate);
+		if (ret != 0) {
+			goto end;
+		}
+		data->dev_cfg.data_rate = dev_cfg->data_rate;
+	}
+
+	if ((param_mask & MSPI_DEVICE_CONFIG_CPP) != 0) {
+		ret = mspi_stm32_xspi_validate_cpp(dev_cfg->cpp);
+		if (ret != 0) {
+			goto end;
+		}
+		data->dev_cfg.cpp = dev_cfg->cpp;
+	}
+
+	mspi_stm32_assign_cfg(data, param_mask, dev_cfg, cfg);
+end:
+	return ret;
+}
+
+/**
+ * API implementation of mspi_dev_config : controller device specific configuration
+ *
+ * @param controller Pointer to the device structure for the driver instance.
+ * @param dev_id Pointer to the device ID structure from a device.
+ * @param param_mask Macro definition of what to be configured in cfg.
+ * @param dev_cfg The device runtime configuration for the MSPI controller.
+ *
+ * @retval 0 if successful.
+ * @retval -EINVAL invalid capabilities, failed to configure device.
+ * @retval -ENOTSUP capability not supported by MSPI peripheral.
```

Comment:
There seems to be other possible return value. Mentioning `A negative errno value upon failure` would maybe be simpler.

This comment applies to other function inline description comments in this P-R.